common/src/main/java/bisq/common/file/JsonFileManager.java

```diff
@@ -76,10 +79,26 @@ public void shutDown() {
     }
 
     public void writeToDiscThreaded(String json, String fileName) {
-        getExecutor().execute(() -> writeToDisc(json, fileName));
+        getExecutor().execute(() -> write(json, fileName));
     }
 
-    public void writeToDisc(String json, String fileName) {
+    public Optional<String> read(String fileName) {
+        File jsonFile = new File(Paths.get(dir.getAbsolutePath(), fileName + ".json").toString());
+        if (!jsonFile.exists()) {
+            return Optional.empty();
+        }
+        StringBuilder json = new StringBuilder();
+        try (Scanner scanner = new Scanner(jsonFile)) {
+            while (scanner.hasNext()) {
+                json.append(scanner.next());
+            }
+        } catch (FileNotFoundException e) {
+            e.printStackTrace();
+        }
```

Comment:
Shouldn't line 96 be `log.error("", e);`  if not a thrown exception?  Stack traces printed to std.err can be out of sync with the logger's buffer flushing -- it can make the timing of the stack trace in the bisq.log file a bit confusing.